### PR TITLE
[TRI-31855] limiting github actions runs to specific paths

### PR DESCRIPTION
**Pull request by shilpasivanesan-tri**
Github actions to run only for backend changes
mac os image test is removed

### CODE REVIEW

**Comment by shilpasivanesan-tri** [comment]
**Validated that the github actions gets triggered only when files under piro folder is updated**

**1 UI changes - only web testing and linting runs**
Screenshot 2024-02-02 at 3 46 58 PM

**2 piro files changes - all the actions runs**
Screenshot 2024-02-02 at 3 48 12 PM

cc @mike-puzon-tri

**Comment by shilpasivanesan-tri** [comment]
1. `piro-test-pr` didnt run as this PR has changes only with github workflow files 

2. `piro-test-main` ran as I have reverted all the validation changes added to files under piro folder in the last commit

File: .github/workflows/piro-test-main.yml [hunk 1]
```diff
@@ -1,6 +1,9 @@
 name: Testing - main
 
-on: [push]
+on:
+  push:
+    paths:
+      - 'piro/**/*'
```
Comment (mike-puzon-tri) [inline .github/workflows/piro-test-main.yml:6]: I think `web/backend` should be added.

**Review by mike-puzon-tri — CHANGES_REQUESTED** [review]

File: .github/workflows/piro-test-main.yml (hunk 1 above)
Comment (mike-puzon-tri) [reply .github/workflows/piro-test-main.yml:6]: @ChrisFajardo-TRI please review if this is correct.

Comment (ChrisFajardo-TRI) [reply .github/workflows/piro-test-main.yml:6]: no need for web/backend. we don't have tests for that, the tests are for the Piro folder

**Review by ChrisFajardo-TRI — APPROVED** [review]

**Pull request merged**